project/src/hooks/useMap.tsx

```diff
@@ -4,36 +4,39 @@ import {OpenStreetMapProvider} from 'leaflet-geosearch';
 
 function useMap (mapRef: MutableRefObject<HTMLElement | null>, city: string): Map | null {
   const [map, setMap] = useState<Map | null>(null);
-  const provider = new OpenStreetMapProvider();
+  const newLocal = new OpenStreetMapProvider();
+  const provider = newLocal;
 
   useEffect(() => {
-    provider.search({ query: `${city} `}).then((result) => {
-      const lat = parseFloat(result[0].raw.lat);
-      const lng = parseFloat(result[0].raw.lon);
-      return [lat, lng];
-    }).then((point) => {
-      if (mapRef.current !== null && map === null) {
-        const instance = new Map(mapRef.current, {
-          center: {
-            lat: point[0],
-            lng: point[1],
-          },
-          zoom: 10,
-        });
+    provider.search({ query: `${city} `})
```

Comment:
```suggestion
    provider.search({ query: city})
```

city же и так строка, зачем еще раз в строку оборачивать?